File: 2_JAVA/Assignment 5/PowerSequence.java

package Sequence;
final class PowerSequence extends Sequence implements Resetable{
	private double ratio;
	private double current;

	public PowerSequence(){
		ratio = 3;
		current =1;
	}

	public double next(){
		double term = current;
                current += ratio;
		return term;
	}
	public void reset(){
		current = 1;
	}
}

File: Assignment4/DemoApp/powersequence.cs
namespace Met.Sequence;
sealed class PowerSequence : Sequence,IResetable
{
  private double ratio = 3;
  private double current = 1;

  public override double Next()
  {
    double term = current;
    current *= ratio;
    return term;
  }
  public void Reset()
  {
    current = 1;

  }

  }

